dj/models/history.py

```diff
@@ -0,0 +1,40 @@
+"""
+Model for history.
+"""
+from datetime import datetime, timezone
+from enum import Enum
+from functools import partial
+from typing import Any, Dict, Optional
+
+from sqlalchemy import DateTime
+from sqlalchemy.sql.schema import Column as SqlaColumn
+from sqlmodel import Field, JSON, SQLModel
+
+from dj.typing import UTCDatetime
+
+class EntityType(str, Enum):
+    attribute = "attribute"
+    catalog = "catalog"
+    engine = "engine"
+    namespace = "namespace"
+    node = "node"
+    query = "query"
+    tag = "tag"
+
+class Activity(SQLModel, table=True):  # type: ignore
```

Comment:
@shangyian @CircArgs @betodealmeida @agorajek @cwegener let me know what you all think about the attributes on this `Activity` class. Are there any other things we'd want to capture at the top level for individual activity events? I've included a JSON `details` attribute for capturing dynamic pieces of metadata but anything that we'd want to expose a filter on should probably be pulled out as a first class attribute.

I'm thinking the history endpoint would allow filtering on `entity_type`, i.e. `entity_type=EntityType.node` and `entity_name`, i.e. `entity_name=basic.num_users`.

Reply:
I think this looks like a good start! Do we want to capture some kind of diff of what changed (perhaps represented by json in `details`) by default? That way we'd be capturing all the potential information we need for the future, in case there's something we later decide is important (and we can pull it out as a first-class attribute then).